plot_antennas_iq

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# Author: Remington Rohel
+
+# This script gets the latest files from
+
+# Copied from https://www.smashingmagazine.com/2015/06/efficient-image-resizing-with-imagemagick/#bash-shell
+smartresize() {
+  # Takes 3 args: input image, desired width (px), and output_directory. Scales and compresses image, maintaining aspect ratio.
+  mogrify -path "$3" -filter Triangle -define filter:support=2 -thumbnail "$2" -unsharp 0.25x0.08+8.3+0.045 -dither None -posterize 136 -quality 82 -define jpeg:fancy-upsampling=off -define png:compression-filter=5 -define png:compression-level=9 -define png:compression-strategy=1 -define png:exclude-chunk=all -interlace none -colorspace sRGB "$1"
+}
+
+echo "Running plot_antennas_iq at $(date -u +%Y%m%d.%H%M)"
+now=$(date -u +%s)  # seconds since epoch
+
+# Find out when the previous 2-hr window started
+# search_1 = YYYYMMDD.HH for 2 hours ago
+search_1=$(date -d @$((now - 7200)) +%Y%m%d.%H)
+
+# Find the previous hour
+# search_2 = YYYYMMDD.HH for 1 hour ago
+search_2=$(date -d @$((now - 3600)) +%Y%m%d.%H)
+
+# Important directories for data and plots
+data_dir=/borealis_nfs/borealis_data/antennas_iq_array/
+failed_dir=/borealis_nfs/borealis_data/conversion_failure/
+plot_dir=/home/transfer/logging/daily_plots/
+fullsize_dir=/home/transfer/logging/daily_plots/fullsize/
+
+# Check the existence of the necessary directories.
+if [ ! -d "${plot_dir}" ]; then
+    echo "Making directory ${plot_dir}"
+    mkdir "${plot_dir}"
+fi
+
+if [ ! -d "${fullsize_dir}" ]; then
+    echo "Making directory ${fullsize_dir}"
+    mkdir "${fullsize_dir}"
+fi
+
+# Get the files from the past two hours (matching search_1 or search_2 in the data_dir and failed_dir)
+daily_files=$(find "${data_dir}" "${failed_dir}" -type f -regex ".*\(${search_1}\|${search_2}\).*0.antennas.*")
+
+echo "${daily_files}"
+
+# Need pydarnio-env to plot files
+source /home/transfer/pydarnio-env/bin/activate
+
+for f in $daily_files; do
+  python3 /home/transfer/borealis-data-utils/SNR/plot_antennas_range_time.py "${f}" --plot-directory=${fullsize_dir} --num-processes=1
+done
+
+# Can deactivate pydarnio-env here
+deactivate
+
+# Get the names of all plots just generated
+daily_plots=$(ls ${fullsize_dir}/*.{png,jpg})
```

Comment:
Looks like if an image resize fails, the image won't be removed?
In that case, daily_plots will continually retry to smartresize the images that previously failed.

Not sure if you want that behaviour or not - maybe add a message or move the failed resize images to another directory? Or daily_plots=$(ls ${fullsize_dir}/${search1|search2}.{png,jpg}) <- not the right format but you get the idea, only grab plots from the past two hours

Reply:
I think moving to a separate directory is easiest. If it fails once it'll likely fail twice, so better to just put them aside.